core/src/apps/management/apply_settings.py

```diff
@@ -22,50 +22,55 @@ async def apply_settings(ctx: wire.Context, msg: ApplySettings):
         and msg.passphrase_always_on_device is None
         and msg.display_rotation is None
         and msg.auto_lock_delay_ms is None
+        and msg.unsafe_prompts is None
     ):
         raise wire.ProcessError("No setting provided")
 
     if msg.homescreen is not None:
         if len(msg.homescreen) > storage.device.HOMESCREEN_MAXSIZE:
             raise wire.DataError("Homescreen is too complex")
         await require_confirm_change_homescreen(ctx)
+        try:
+            storage.device.set_homescreen(msg.homescreen)
+        except ValueError:
+            raise wire.DataError("Invalid homescreen")
 
     if msg.label is not None:
         await require_confirm_change_label(ctx, msg.label)
+        storage.device.set_label(msg.label)
 
     if msg.use_passphrase is not None:
         await require_confirm_change_passphrase(ctx, msg.use_passphrase)
+        storage.device.set_passphrase_enabled(msg.use_passphrase)
 
-    if msg.passphrase_always_on_device is not None:
+    if (
+        storage.device.is_passphrase_enabled()
+        and msg.passphrase_always_on_device is not None
+    ):
```

Comment:
I guess this is correct but we should inform the user via wire.Error in case the passphrase is not enabled.

Reply:
df1fa74